### ISSUE TITLE
Recent italics support seems cause rendering issue with bold and reverseVideo

### ISSUE THREAD

**Issue by glguy** [jtdaugherty/vty#155]
I'm testing this on the built-in Mac terminal.

Consider this program. Somehow it causes the text that should be bold to be invisible, and the second reverse video B to be flashing instead.

```haskell
module Main where

import Graphics.Vty
import Control.Exception

main =
  bracket (mkVty mempty) shutdown $ \vty ->
    do update vty (picForImage image)
       nextEvent vty

image = string (defAttr `withStyle` reverseVideo) "B"    <|>
        string (defAttr `withStyle` bold        ) "bold" <|>
        string (defAttr `withStyle` reverseVideo) "B"
```

**Comment by glguy** [comment]
https://github.com/jtdaugherty/vty/blob/181a5d77df911c3d510f11043a5af7b1d62493dd/src/Graphics/Vty/Output/TerminfoBased.hs#L452

this line is related, removing it fixes the bug. I'm still working out what the bug *is*

I looks like this element should only be in the list if italics are supported. Currently turning on italics causes reverse video effect to be applied in isolation.


My reading of https://invisible-island.net/ncurses/man/terminfo.5.html suggests that italics aren't set by SGR and that this line should be removed, and that we need a separate way to set the italic mode when necessary.

**Comment by glguy** [comment]
https://github.com/glguy/vty/commit/551e1431082b7907eec48c04693def32c5bc746a working toward a fix. Italics still partially work under this model.

**Comment by jtdaugherty** [comment]
Thanks for discovering/investigating this!

**Comment by jtdaugherty** [comment]
Does the program you posted above misbehave on 5.22? I don't see how adding italics would have impacted that program.

**Comment by glguy** [comment]
Yeah, the bug is new in 5.23.  adding italics causes the bug by sending an invalid SGR mode

**Comment by glguy** [comment]
I think the fix is to restore the old SGR mode command by deleting the line I linked to above and then always using the start and stop control sequences when italics are involved in a style diff

**Comment by glguy** [comment]
I'm going to exclude vty-5.23 from my build-depends sections, but we should probably revise it to be unbuildable by setting the base range to `<0` on hackage so that other packages don't pull it in.

**Comment by glguy** [comment]
As a point of information, the reason I discovered this (I've learned) is that there's no sequence for turning off _reverse video_, you have to reset the attributes altogether to do that via the SGR command. Italics were working on the more common isolated case.

**Comment by glguy** [comment]
Fixed in PR

**Comment by jtdaugherty** [comment]
Thanks, @glguy!

**Comment by jtdaugherty** [comment]
CC @milesrout

**Comment by jtdaugherty** [comment]
(Fix released in `5.23.1`.)
